File: fc-chat/demo/part3/ch3/02-6-message-system-add-connection/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/RejectRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;

public class RejectRequest extends BaseRequest {

  private final String username;

  @JsonCreator
  public RejectRequest(@JsonProperty("username") String username) {
    super(MessageType.REJECT_REQUEST);
    this.username = username;
  }

  public String getUsername() {
    return username;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/RejectRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType


class RejectRequest @JsonCreator constructor(
    @param:JsonProperty(
        "username"
    ) val username: String
) :
    BaseRequest(MessageType.REJECT_REQUEST)
